### PR TITLE
DeviceEventFactory: Modified event factory to provide correct value for buttons property

### PR DESCRIPTION
**Pull request by PolygonalSun**
There was an [issue](https://github.com/BabylonJS/Babylon.js/issues/13920) brought up where drag controls where not "letting go" when the mouse button was released.  This PR will now properly propagate this value based off of the MouseEvent standard.

Edit: This also contains a little change to out pointer capture is being handled.  Basically, if the POINTERDOWN is skipped, the pointer capture bool is still set to true and if the POINTERUP is skipped and there are no buttons pressed, it should be released.

### CODE REVIEW

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/13927/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13927/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13927/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13927/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13927/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13927/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13927/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13927/merge#BCU1XR#0

**Review by sebavan — APPROVED** [review]

File: packages/dev/core/src/DeviceInput/eventFactory.ts [hunk 1]
```diff
@@ -76,6 +76,15 @@ export class DeviceEventFactory {
             evt.pointerType = "touch";
         }
 
+        let buttons = 0;
+
+        // Populate buttons property with current state of all mouse buttons
+        // Uses values found on: https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/buttons
+        buttons += deviceInputSystem.pollInput(deviceType, deviceSlot, PointerInput.LeftClick);
```
Comment (RaananW) [inline packages/dev/core/src/DeviceInput/eventFactory.ts:83]: just making sure - pollInput returns 0 or 1?

**Review by RaananW — COMMENTED** [review]
Seems like the input tests are failing.

**Comment by PolygonalSun** [comment]
> Seems like the input tests are failing.

Yeah, I found an issue with a bool in InputManager, I've pushed a fix for that.

File: packages/dev/core/src/DeviceInput/eventFactory.ts (hunk 1 above)
Comment (PolygonalSun) [reply packages/dev/core/src/DeviceInput/eventFactory.ts:83]: Yes, that is correct.

**Pull request merged**